request/private/http-location.rkt

```diff
@@ -7,18 +7,51 @@
 
 (provide
  (contract-out
-  [make-domain-requester (-> string? requester? requester?)]
-  [make-host+port-requester (-> string? exact-nonnegative-integer? requester? requester?)]))
+  [make-domain-requester
+   (-> string? requester? requester?)]
+  [make-host+port-requester
+   (-> string? exact-nonnegative-integer? requester? requester?)]
+  [make-https-requester
+   (-> string? requester? requester?)]))
 
 
-(define (domain+relative-path->http-url domain relative-path)
-  (string->url (format "http://~a/~a" domain relative-path)))
+(define (domain+relative-path->http-url protocol domain relative-path)
+  (string->url (format "~a://~a/~a" protocol domain relative-path)))
 
 (define (host+port->domain host port)
   (format "~a:~a" host port))
 
 (define (make-domain-requester domain requester)
-  (wrap-requester-location (domain+relative-path->http-url domain _) requester))
+  (wrap-requester-location
+   (domain+relative-path->http-url "http" domain _) requester))
+
+(define (make-https-requester domain requester)
+  (wrap-requester-location
+   (domain+relative-path->http-url "https" domain _) requester))
 
 (define (make-host+port-requester host port requester)
   (make-domain-requester (host+port->domain host port) requester))
+
+(module+ test
+  (require rackunit
+           "base.rkt"
+           "call-response.rkt")
+
+  (define domain "httpbin.org")
+  (define http-url (domain+relative-path->http-url "http" domain "/"))
+  (define https-url (domain+relative-path->http-url "https" domain "/"))
+
+  (define http-req (make-domain-requester domain http-requester))
+  (define https-req
+    (make-https-requester domain http-requester))
+ 
+  (define http-resp (get http-req "/ip"))
```

Comment:
This test will fail in the Racket catalog builds because they're performed in an isolated VM without network access. I'd recommend changing this from `(module+ test ...)` to `(module+ integration-test ...)` and adding a line to the Travis script to run `raco test` on the integration test submodule.